aten/src/ATen/native/native_functions.yaml

```diff
@@ -3404,8 +3404,13 @@
   requires_tensor: True
 
 
+- func: to_custom(Tensor self, str type) -> Tensor
+  variants: method
```

Comment:
Even considering this API on its merits, it seems kind of strange to me that this has to take a Tensor to create a custom tensor. What exactly am I feeding it here, anyway?

Reply:
Conceptually, tensors are well represented by specifications of indexing, dtype and value. The underlying storage, device and associated method implementations are irrelevant.  `to_custom` swaps the latter bit without changing the former.

Reply:
dumb question: why do we need it?  I.e. why can't someone write a per-custom-type extension function for doing this?